Building a Time Series Weather Forecasting Application in Python (#5736)
section-engineering-education/engineering-education: content/articles/building-a-time-series-weather-forecasting-application-in-python/index.md

    alt: Time Series Weather Forecasting Application Hero image
---
This tutorial will look at how we can forecast the weather using a time series package known as Neural Prophet. 
<!--more-->
In this walkthrough, we will be going through a couple of key things:
- We'll start by preprocessing our data fetched from Kaggle using the Pandas library.
- We'll train a time series forecasting model to predict temperature using the model.
- We'll learn how to forecast the temperature into the future.

### Prerequisites
- You can use either Google Colab or Jupyter Notebook.
> To follow along, please use Google Colab.
- You need to be familiar with Machine Learning modeling. 

### Table of contents
- [About Neural Prophet](#about-neural-prophet)
- [Installing and importing the required dependencies](#installing-and-importing-the-required-dependencies)
- [Loading the dataset](#loading-the-dataset)
- [Preprocessing the data ](#preprocessing-the-data )
- [Training the forecasting model]((#training-the-forecasting-model))
- [Forecasting the temperature into the future](#forecasting-the-temperature-into-the-future)
- [Wrapping up](#wrapping-up)

### About Neural Prophet

> It is a time-series model that has been built on top of [AR-Net](https://ai.facebook.com/blog/ar-net-a-simple-autoregressive-neural-network-for-time-series/) and [Facebook Prophet](https://github.com/facebook/prophet). It is an upgraded version of Facebook Prophet. It uses the PyTorch framework as a backend. It is beginner friendly and one can get started using it a quick `pip` install. It incorporates traditional statistical and neural network models for time series modeling, used in forecasting and anomaly detection. The model generates high quality forecasts for time series data that has multiple seasonality with linear or non-linear growth.

Reviewer: ```suggestion
It is a time-series model that has been built on top of [AR-Net](https://ai.facebook.com/blog/ar-net-a-simple-autoregressive-neural-network-for-time-series/) and [Facebook Prophet](https://github.com/facebook/prophet). It is an upgraded version of Facebook Prophet. It uses the PyTorch framework as a back-end. It is beginner-friendly and one can get started using a quick `pip` install. It incorporates traditional statistical and neural network models for time series modeling, used in forecasting and anomaly detection. The model generates high-quality forecasts for time series data that have multiple seasonality with linear or non-linear growth.
```

```python
import pandas as pd
from neuralprophet import NeuralProphet
from matplotlib import pyplot as plt
```
- `Pandas` will help us read our data into our notebook.
- `NeuralProphet` is the class we will use to help us predict the future temperature.
- `Matplotlib` will help us in plotting.

The next step involves us importing our data.

### Loading the dataset 
We will use the [Austin Weather](https://www.kaggle.com/grubenm/austin-weather) dataset from Kaggle. Although it is a dataset containing the historical temperature, precipitation, humidity, and windspeed for Austin, Texas, we will only predict the temperature. That means we will only work with the temperature data from the dataset. You need to download it and upload the `austin_weather.csv` file into your notebook.

```python
df = pd.read_csv('austin_weather.csv')
df.tail()
```
We have used Pandas `read_csv()` method to load our dataset. In addition, we've used the `tail()` method to view the last five rows in our dataset.

Let us do a bit of exploratory data analysis on the data.

```python
df.Date.unique()
```

> When you run the code above, you'll see that the dates in our dataset that'll be used for training range between `2013-12-21` and `2017-07-31`. That's about four years worth of data. 

Reviewer: ```suggestion
When you run the code above, you'll see that the dates in our dataset that'll be used for training range between `2013-12-21` and `2017-07-31`. That's about four years' worth of data. 
```

       'VisibilityAvgMiles', 'VisibilityLowMiles', 'WindHighMPH', 'WindAvgMPH',
       'WindGustMPH', 'PrecipitationSumInches', 'Events'],
      dtype='object')
```

As we advance, we will only be focusing on the `TempAvgF` column.

Let's now do a bit of preprocessing.

### Preprocessing the data 
We begin by checking the data types of the columns.

```python
df.dtypes
```
Output:

```bash
Date                          object
TempHighF                      int64
TempAvgF                       int64
TempLowF                       int64
DewPointHighF                 object
DewPointAvgF                  object
DewPointLowF                  object
HumidityHighPercent           object
HumidityAvgPercent            object
HumidityLowPercent            object
SeaLevelPressureHighInches    object
SeaLevelPressureAvgInches     object
SeaLevelPressureLowInches     object
VisibilityHighMiles           object
VisibilityAvgMiles            object
VisibilityLowMiles            object
WindHighMPH                   object
WindAvgMPH                    object
WindGustMPH                   object
PrecipitationSumInches        object
Events                        object
dtype: object
```

> We'll need to change the `Date` format from an `object` to a `Date time` format. This is because the algorithm only accepts the `date-time` format for the date column. 

Reviewer: ```suggestion
We will need to change the `Date` format from an `object` to a `Date time` format. The algorithm only accepts the `date-time` format for the date column.
```
